### PR TITLE
implement Author, Rent to project

### PR DESCRIPTION
**Pull request by tarasulo**

### CODE REVIEW

File: src/main/java/library/app/spring/dao/impl/AuthorDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        TypedQuery<Author> query = sessionFactory.getCurrentSession()
+                .createQuery("from Author where name:=name");
+        query.setParameter("name", name);
+        return query.getResultList();
+    }
+
+    @Override
+    public List<Author> findByNameAndSurname(String name, String surname) {
+        TypedQuery<Author> query = sessionFactory.getCurrentSession()
+                .createQuery("from Author where name:=name and surname:=surname");
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/dao/impl/AuthorDaoImpl.java]: better use `like` in query

File: src/main/java/library/app/spring/entity/Rent.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "rent_date")
+    private LocalDate rentDate;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id")
+    private User user;
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/entity/Rent.java]: Why do we use many to many here?
maybe we should use one to one?

Comment (tarasulo) [reply src/main/java/library/app/spring/entity/Rent.java]: make sense, done

File: src/main/java/library/app/spring/dao/impl/AuthorDaoImpl.java (hunk 1 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/dao/impl/AuthorDaoImpl.java]: done

File: src/main/java/library/app/spring/dao/impl/UserDaoImp.java [hunk 3]
```diff
@@ -23,7 +23,8 @@ public void add(User user) {
     @Override
     public List<User> listUsers() {
         @SuppressWarnings("unchecked")
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/dao/impl/UserDaoImp.java]: try to remove `SuppressWarnings`. Bohdan wrote how to do it in slack

File: src/main/java/library/app/spring/entity/Rent.java [hunk 4]
```diff
@@ -24,11 +22,11 @@
     @Column(name = "rent_date")
     private LocalDate rentDate;
 
-    @ManyToOne(fetch = FetchType.LAZY)
+    @OneToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "user_id")
     private User user;
 
-    @OneToOne(cascade = CascadeType.ALL)
+    @OneToOne(fetch = FetchType.LAZY)
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/entity/Rent.java]: why we use lazy here?

File: src/main/java/library/app/spring/entity/Rent.java [hunk 5]
```diff
@@ -24,11 +22,11 @@
     @Column(name = "rent_date")
     private LocalDate rentDate;
 
-    @ManyToOne(fetch = FetchType.LAZY)
+    @OneToOne(fetch = FetchType.LAZY)
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/entity/Rent.java]: the same

File: src/main/java/library/app/spring/entity/Author.java [hunk 6]
```diff
@@ -25,7 +27,10 @@
     @Column(name = "surname")
     private String surname;
 
-    @ManyToMany(mappedBy = "authors", fetch = FetchType.EAGER)
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (Sofasmile) [inline src/main/java/library/app/spring/entity/Author.java]: why we use `join table` in author and book models?

File: src/main/java/library/app/spring/dao/impl/UserDaoImp.java (hunk 3 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/dao/impl/UserDaoImp.java]: done

File: src/main/java/library/app/spring/entity/Rent.java (hunk 4 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/entity/Rent.java]: agree, used EAGER

File: src/main/java/library/app/spring/entity/Rent.java (hunk 5 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/entity/Rent.java]: done

File: src/main/java/library/app/spring/entity/Author.java (hunk 6 above)
Comment (tarasulo) [reply src/main/java/library/app/spring/entity/Author.java]: change on mapped

**Pull request merged**